iterators task (1)/student/include/impl.h

```diff
@@ -5,11 +5,59 @@ template <class Container>
 struct accumulator
 {
     /** @todo Iterator traits*/
+    typedef std::output_iterator_tag            iterator_category;
+    typedef typename Container::value_type      value_type;
+    typedef typename Container::difference_type difference_type;
+    typedef typename Container::pointer         pointer;
+    typedef typename Container::reference       reference;
+
     /** @todo Constructor with container*/
+    explicit accumulator(Container& _container )
+        :_container_ref(_container)
+    {}
+
     /** @todo Copy assignment operator*/
+    accumulator& operator=( const Container& _container)
+    {
+        _container_ref = _container;
+        return *this;
+    }
+
+    accumulator& operator=( const reference _value)
+    {
+        _container_ref.push_back(_value);
+        return *this;
+    }
+
     /** @todo Move assignment operator*/
+    accumulator& operator=( const Container&& _container )
```

Comment:
**this type is not valid!
when you trying to use move on const rvalue&& it collapses to const lvalue&
don't use rvalue and const together!**